### PR TITLE
update SSO integration instructions (Microsoft Entra ID)

### PR DESCRIPTION
**Pull request by khorne3**
We have several fixes in this PR:

* [Renamed all instances of Azure Active Directory to Microsoft Entra ID](https://linear.app/semgrep/issue/MKT-779/change-mentions-of-active-directory-azure-ad-to-microsoft-entra-id)
* Updated instructions and screenshots so that it's accurate (I tested the SSO configuration myself, and it works as detailed in this updated PR)
* [Fixed error with attribute mapping](https://linear.app/semgrep/issue/MKT-780/ensure-user-attribute-mapping-is-correct-for-azure-ad-entra-id)

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team

### CODE REVIEW

File: docs/deployment/sso.md [hunk 1]
```diff
@@ -80,59 +80,66 @@ The following references are used:
...
+
+You have now created a custom enterprise app for Semgrep to integrate with Microsoft Entra ID. This enables you to set up SAML SSO.
+
+#### Set up SAML SSO for your new enterprise app
+
+1. From your new enterprise app's page, go to **Single-sign on** > **SAML**. 
+   ![Enterprise application's Single-sign on menu option](/img/entra-3.png#md-width)
+2. When prompted to **Select a single sign-on method**, select **SAML**. You are redirected to the **SAML-based Sign-on** page.
+   ![SAML-based Sign-on screen](/img/entra-4.png#md-width)
+3. Click **Edit** to update your **Basic SAML Configuration** by providing the **Entity ID** and **Reply URL**. You can retrieve these values from Semgrep Cloud Platform by performing the following steps:
```
Comment (s-santillan) [inline docs/deployment/sso.md:114]:
```suggestion
3. In the **Basic SAML Configuration** section, click **Edit**. Provide the **Entity ID** and **Reply URL**. You can retrieve these values from Semgrep Cloud Platform by performing the following steps:
```

File: docs/deployment/sso.md [hunk 2]
```diff
@@ -80,59 +80,66 @@ The following references are used:
...
+    1. Log in to Semgrep Cloud Platform and navigate to [**Settings** > **Access** > **SSO**](https://semgrep.dev/orgs/-/settings/access/sso) page. 
+    2. Click **Add SAML2 SSO**.
+    3. Copy the **Audience URL** value from Semgrep Cloud Platform. Return to **Basic SAML Configuration**. Click **Add identifier** to paste this value as the **Identifier (Entity ID)**.
+    4. Copy the **SSO URL** value from Semgrep Cloud Platform. Return to **Basic SAML Configuration**. Click **Add reply URL** to paste this value as the **Reply URL (Assertion Consumer Service URL)**.
+4. Click **Save** and close out of **Basic SAML Configuration**.
+5. Click **Edit** to modify **Attributes & Claims**. You must add two claims. To add your first claim:
```
Comment (s-santillan) [inline docs/deployment/sso.md:120]:
```suggestion
5. In the **Attributes and Claims** section, click **Edit**. You must add two claims. To add your first claim:
```

File: docs/deployment/sso.md [hunk 3]
```diff
@@ -80,59 +80,66 @@ The following references are used:
...
+    1. Click **Add new claim**.
+    2. Enter `email` in the **Name** field.
+    3. From the **Source attribute** drop-down box, select `user.mail`.
+    4. Click **Save**.
+7. Close out of **Attributes & Claims**.
+8. Navigate to Semgrep Cloud Platform, and provide the values required by the SAML2 form:
+    1. Provide the **Display name** and the **Email domain** you are using for the integration.
+    2. Copy the **Login URL** value from Microsoft Entra ID and paste it in into SCP's **IDP SSO URL** field.
+    3. Copy and paste the **Microsoft Entra ID Identifier** value into SCP's **IdP Issuer ID** field.
+    4. In Entra ID's **SAML-based Sing-on** page, click **Download** to obtain the **Certificate (Base64)**.
```
Comment (s-santillan) [inline docs/deployment/sso.md:135]:
```suggestion
    4. In Entra ID's **SAML-based Sign-on** page, click **Download** to obtain the **Certificate (Base64)**.
```

**Review by s-santillan — APPROVED** [review]
Only 2 ignorable nits :) Thank you for this update.

File: docs/deployment/sso.md [hunk 4]
```diff
@@ -80,59 +80,66 @@ The following references are used:
...
+3. In the **Basic SAML Configuration** section, click **Edit**. Provide the **Entity ID** and **Reply URL**. You can retrieve these values from Semgrep Cloud Platform by performing the following steps:
+    1. Log in to Semgrep Cloud Platform and navigate to [**Settings** > **Access** > **SSO**](https://semgrep.dev/orgs/-/settings/access/sso) page. 
+    2. Click **Add SAML2 SSO**.
+    3. Copy the **Audience URL** value from Semgrep Cloud Platform. Return to **Basic SAML Configuration**. Click **Add identifier** to paste this value as the **Identifier (Entity ID)**.
+    4. Copy the **SSO URL** value from Semgrep Cloud Platform. Return to **Basic SAML Configuration**. Click **Add reply URL** to paste this value as the **Reply URL (Assertion Consumer Service URL)**.
+4. Click **Save** and close out of **Basic SAML Configuration**.
+5. In the **Attributes and Claims** section, click **Edit**. You must add two claims. To add your first claim:
+    1. Click **Add new claim**.
+    2. Enter `name` in the **Name** field.
+    3. For the **Source attribute** drop-down box, select `user.givenname`.
```
Comment (armchairlinguist) [inline docs/deployment/sso.md:123]: I know this is what we had before, but I wonder if we should actually be using `user.displayName`? The given name is just the first name which in many orgs is not enough to properly identify a user.

**Review by armchairlinguist — COMMENTED** [review]
Seems sensible. Our real SME on this is out of the office, alas.

**Comment by khorne3** [comment]
I'm pretty comfortable w/ the instructions as-is, since they worked during testing. I'll merge, and we can update with anything clarifying at a later date (as it is, things are NOT correct).

**Pull request merged**